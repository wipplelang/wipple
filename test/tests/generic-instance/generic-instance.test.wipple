-- [should compile]

Box : A => type {value :: A}

Foo : A B => trait (A -> B)

A => instance (Foo (Box A) A) : {value : value} -> value

show (Foo {value : "Hello, world!"})
