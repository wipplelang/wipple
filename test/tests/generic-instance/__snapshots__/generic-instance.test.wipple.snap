// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generic-instance.test.wipple 1`] = `[]`;

exports[`generic-instance.test.wipple 2`] = `
{
  "output": "Hello, world!
",
}
`;
